Rename file and module

KasperJanssens opened the pull request:
Part two of the chopped up PR

carymrobbins commented:
Merged!  This looks good as well.  It might also be nice to rename the full qualified name as well and move the file accordingly, but this should cover most cases.

KasperJanssens commented:
That kind of thing is normally a move. Come to think of it, I'm not sure that I implemented this case in my move branch (I implemented move in the mean time as well, will offer it up one of these days, it's a bit dependent on the cabal branch, as moving will also rename the cabal dependency. The cabal branch itself will include an extended rename that will also change the the cabal dependency when renaming.)
